File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    let p1 = [1,2,3,4,5];
    let p2 = [2, 1, 2, 3, 2, 4, 2, 5];
    let p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    let count =[0,0,0]
   
    for (let i =0; i< answers.length; i++){
        if(answers[i] === p1[i%p1.length]){
            count[0]+=1
        }
        if(answers[i] === p2[i%p2.length]){
            count[1]+=1
        }
        if(answers[i] === p3[i%p3.length]){
            count[2]+=1
        }
    }
    
    let maxValue = Math.max(...count);
    let answer = []
    
    for (let i=0; i<3; i++){
        if(maxValue === count[i]){
            answer.push(i+1);
        }
    }
    
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[] math1 = {1,2,3,4,5};
        int[] math2 = {2,1,2,3,2,4,2,5};
        int[] math3 = {3,3,1,1,2,2,4,4,5,5};
        
        int[] scores = {0,0,0};
        
        //정답 체크
        for(int i =0; i< answers.length; i++){
            int nowNumber = answers[i];
            
            if(nowNumber == math1[i%math1.length]){
                scores[0] +=1;
            }
            if(nowNumber == math2[i%math2.length]){
                scores[1] +=1;
            }
            if(nowNumber == math3[i%math3.length]){
                scores[2] +=1;
            }
        }
        
        //최대 값 체크
        int maxScore = Arrays.stream(scores).max().getAsInt();
        ArrayList<Integer> answerData = new ArrayList<>();
        
        for(int i =0; i< scores.length; i++){
            if(maxScore == scores[i]){
                answerData.add(i+1);
            }
        }
        
         
        return answerData.stream().mapToInt(Integer::intValue).toArray();
    }
}